From d5986924176526d4db4e2505ab516bf1796f92df Mon Sep 17 00:00:00 2001
From: "Richard A. Griffiths" <richard.a.griffiths@intel.com>
Date: Fri, 21 Aug 2015 14:16:23 -0700
Subject: [PATCH 10/17] input: touchscreen: silead: fix unwanted key repeat

silead_ts_report_touch needed an added call to input_report_abs
after the call to input_mt_slot. Also matched the original vendor
driver's subtracting 1 from the id when passing to input_mt_slot

Change-Id: I48ac4dc74dea183f000e6e9e793afe04702257b2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Richard A. Griffiths <richard.a.griffiths@intel.com>
Reviewed-on: https://android.intel.com:443/472783
---
 drivers/input/touchscreen/silead.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 3c3afb3..1f8855c 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -148,7 +148,8 @@ static int silead_ts_request_input_dev(struct silead_ts_data *data)
 static void silead_ts_report_touch(struct silead_ts_data *data, u16 x, u16 y,
 				   u8 id)
 {
-	input_mt_slot(data->input_dev, id);
+	input_mt_slot(data->input_dev, id - 1);
+	input_report_abs(data->input_dev,ABS_MT_TRACKING_ID, id);
 	input_mt_report_slot_state(data->input_dev, MT_TOOL_FINGER, true);
 	input_report_abs(data->input_dev, ABS_MT_POSITION_X, x);
 	input_report_abs(data->input_dev, ABS_MT_POSITION_Y, y);
-- 
1.9.1

